src/Microsoft.AspNetCore.Razor.Runtime/Runtime/TagHelpers/TagHelperExecutionContext.cs

```diff
@@ -214,6 +214,46 @@ public void AddTagHelperAttribute(string name, object value)
             _allAttributes.Add(name, value);
         }
 
+        // Used to pool TagHelperExecutionContext's. With this we don't always need to recreate TagHelperExecutionContexts.
+        internal void ClearAndUpdateWith(
```

Comment:
Suggest `Reinitialize()`. Also suggest `public` because `internal` is used only for bits that would be `private` if not for testing. (All of `Microsoft.AspNetCore.Razor.Runtime.TagHelpers` could be named `.Internal` if we wanted.)